exporters/otlp/otlptrace/otlptracegrpc/client.go

```diff
@@ -17,94 +17,233 @@ package otlptracegrpc // import "go.opentelemetry.io/otel/exporters/otlp/otlptra
+	cfg := otlpconfig.NewGRPCConfig(asGRPCOptions(opts)...)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	c := &client{
+		endpoint:      cfg.Traces.Endpoint,
+		exportTimeout: cfg.Traces.Timeout,
+		requestFunc:   cfg.RetryConfig.RequestFunc(evaluate),
+		dialOpts:      cfg.DialOptions,
+		stopCtx:       ctx,
+		stopFunc:      cancel,
+		conn:          cfg.GRPCConn,
 	}
 
-	c := &client{}
-	c.connection = connection.NewConnection(cfg, cfg.Traces, c.handleNewConnection)
+	if len(cfg.Traces.Headers) > 0 {
+		c.metadata = metadata.New(cfg.Traces.Headers)
+	}
 
 	return c
 }
 
-func (c *client) handleNewConnection(cc *grpc.ClientConn) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	if cc != nil {
-		c.tracesClient = coltracepb.NewTraceServiceClient(cc)
-	} else {
-		c.tracesClient = nil
+// Start establishes a gRPC connection to the collector.
+func (c *client) Start(ctx context.Context) error {
+	if c.conn == nil {
+		// If the caller did not provide a ClientConn when the client was
+		// created, create one using the configuration they did provide.
+		conn, err := grpc.DialContext(ctx, c.endpoint, c.dialOpts...)
+		if err != nil {
+			return err
+		}
+		c.conn = conn
 	}
-}
 
-// Start establishes a connection to the collector.
-func (c *client) Start(ctx context.Context) error {
-	return c.connection.StartConnection(ctx)
+	// The otlptrace.Client interface states this method is called just once,
+	// so no need to check if already started.
+	c.tscMu.Lock()
+	c.tsc = coltracepb.NewTraceServiceClient(c.conn)
+	c.tscMu.Unlock()
+
+	return nil
 }
 
-// Stop shuts down the connection to the collector.
+var errNotStarted = errors.New("client not started")
+
+// Stop shuts down the gRPC connection to the collector.
+//
+// If the client was crated using WithGRPCConn, the passed conn is closed.
 func (c *client) Stop(ctx context.Context) error {
-	return c.connection.Shutdown(ctx)
+	// Acquire the c.tscMu lock within the ctx lifetime.
+	acquired := make(chan struct{})
+	go func() {
+		c.tscMu.Lock()
+		close(acquired)
+	}()
+	var ctxErr error
+	select {
+	case <-ctx.Done():
+		// The Stop timeout is reached. Kill any remaining exports to force
+		// the clear of the lock and save the timeout error to return and
+		// signal the shutdown timed out before cleanly stopping.
+		c.stopFunc()
+		ctxErr = ctx.Err()
+
+		// To ensure the client is not left in a dirty state c.tsc needs to be
+		// set to nil. To avoid the race condition when doing this, ensure
+		// that all the exports are killed (initiated by c.stopFunc).
+		<-acquired
+	case <-acquired:
+	}
+	// Hold the tscMu lock for the rest of the function to ensure no new
+	// exports are started.
+	defer c.tscMu.Unlock()
+
+	// The otlptrace.Client interface states this method is called only
+	// once, but there is no guarantee it is called after Start. Ensure the
+	// client is started before doing anything and let the called know if they
+	// made a mistake.
+	if c.tsc == nil {
```

Comment:
True, but the [the Client](https://github.com/open-telemetry/opentelemetry-go/blob/034956199a0bef5a1f3e0b81350f51a8fcf8df22/exporters/otlp/otlptrace/clients.go#L37-L41) interface explicitly states it will only ever be called once and that it will be handled at the exporter level.

Reply:
Updated, PTAL.